Userland/Applications/PixelPaint/Image.cpp

```diff
@@ -548,7 +548,7 @@ Optional<Gfx::IntRect> Image::nonempty_content_bounding_rect() const
     return bounding_rect;
 }
 
-void Image::resize(Gfx::IntSize const& new_size, Gfx::Painter::ScalingMode scaling_mode)
+void Image::resize(Gfx::IntSize const& new_size, Gfx::Painter::ScalingMode scaling_mode, bool should_rescale)
```

Comment:
Generally, we prefer to use named enums instead of bools for parameters. Without looking at the function definitions, it's a lot easier to understand something like
```c++
resize(new_size, location(), scaling_mode, ShouldRescale::No);
```
than
```c++
resize(new_size, location(), scaling_mode, false);
```

Thinking about it, you either want to pass a `ScalingMode`, *or* you want to not scale it. Having them be a single parameter would be nice. Maybe a `Gfx::Painter::ScalingMode::None` value, though I'm not sure if that would interfere with other places that use `ScalingMode`.

Reply:
Good point. I've refactored this PR to use a new `ScalingMode` `None` as suggested e9428e4. This doesn't seem to have caused any issues elsewhere. As a result of that, I've also changed the UI for this to include the Resize Canvas option in the radio selections rather than as a separate checkbox to sit better with what's going on internally and not require special handling if the checkbox was toggled back and forth.